### DISCUSSION TITLE
How to use connect/:provider with React Native

### DISCUSSION THREAD

**Discussion by erikfenriz** [strapi/strapi#6972] [Legacy Topics]
We are working with React Native and migrating from Firebase auth to Strapi auth. 
We want to migrate social auth providers as well.
I have already connected Google provider in the admin panel.
![Screenshot 2020-07-10 at 18 28 31](https://user-images.githubusercontent.com/25347909/87171339-3588b700-c2db-11ea-8ce5-18320beaedaa.png)

code: 
```
logInGoogle = async () => {
        try {
            await GoogleSignin.hasPlayServices();
            const userInfo = await GoogleSignin.signIn();
...
```
I get the user's data with id token and credentials from the provider. 
How to use connect/:provider properly in this situation?

**Comment by toanz** [comment]
i think you need to create custom rest api  for authentication plugin with connect provider !

**Comment by erikfenriz** [comment]
We added a Firebase layer in the permissions.js file to handle tokens, authenticating users as usual with Firebase.

**Reply by shtaimoortariq** [reply]
so we would be creating the session with firebase ? i mean our user data would be in firebase ?
